File: CustomChance/Net/Model/CommonLogic.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Collections.ObjectModel;

namespace CustomChance
{
    class CommonLogic
    {
        static CommonLogic _instance;
        
        private CommonLogic()
        {
            Storage = new Storage();
        }

        public static CommonLogic GetInstance()
        {
            if (_instance != null)
                return _instance;
            return _instance = new CommonLogic();
        }

        public Storage Storage { get; set; }

        public void TrySerialize()
        {
            Storage.Serialize();
        }

        public void TryDeserialize()
        {
            Storage.Deserialize();
        }

        public void StartRandom(List<Member> members)
        {
            RandomChoise(members);
            int max = 0;
            int index = 0;
            for (int i = 0; i < members.Count; i++)
            {
                if (max < members[i].Value)
                {
                    index = i;
                    max = members[i].Value;
                }
            }
            members[index].IsWinner = true;
        }

        private void RandomChoise(List<Member> variants)
        {
            //preparation
            if (variants.Count == 0)
                return;
            foreach (var item in variants)
            {
                item.Value = 0;
                item.IsWinner = false;
            }

            //randomizing
            Random rnd = new Random();
            for (int i = 0; i < 100; i++)
            {
                int time = 0;

                while (time == 0)
                    time = DateTime.Now.Millisecond;

                int index = rnd.Next(1, variants.Count * time);
                variants[index / time].Value++;
            }
        }

        public bool AddMember(List<Member> members, String name)
        {
            if (members == null)
                members = new List<Member>();

            if (name.Equals(String.Empty))
                return false;

            foreach (var item in members)
            {
                if (item.Name.Equals(name))
                    return false;
            }
            members.Add(new Member()
            {
                Name = name,
                Value = 0,
                IsWinner = false
            });
            return true;
        }

        public void DeleteMember(List<Member> members, string selected)
        {
            foreach (var item in members)
            {
                if (item.Name.Equals(selected))
                {
                    members.Remove(item);
                    return;
                }
            }
        }

        public void PrintMembersStat()
        {
            foreach (var _ in Storage.Members)
            {
                Console.WriteLine(_.Name + " " + _.Value + " " + _.IsWinner);
            }
        }
    }
}

File: CustomChance/Java/src/main/java/com/customchance/Model/CommonLogic.java
package com.customchance.Model;

import java.util.LinkedList;
import java.util.List;
import java.util.Random;

public class CommonLogic {
    static CommonLogic _instance;

    private CommonLogic() {
        Storage = new Storage();
    }

    public static CommonLogic getInstance() {
        if (_instance != null)
            return _instance;
        return _instance = new CommonLogic();
    }

    public Storage Storage;

    public void trySerialize() {
        Storage.serialize();
    }

    public void tryDeserialize() {
        Storage.deserialize();
    }

    public void startRandom(List<Member> members) {
        randomChoise(members);
        int max = 0;
        int index = 0;
        for (int i = 0; i < members.size(); i++) {
            if (max < members.get(i).value) {
                index = i;
                max = members.get(i).value;
            }
        }
        members.get(index).isWinner = true;
    }

    private void randomChoise(List<Member> variants) {
        // preparation
        if (variants.size() == 0)
            return;
        for (Member item : variants) {
            item.value = 0;
            item.isWinner = false;
        }

        // randomizing
        Random rnd = new Random();
        for (int i = 0; i < 100; i++) {
            int index = rnd.nextInt(variants.size());
            variants.get(index).value++;
        }
    }

    public boolean addMember(List<Member> members, String name) {
        if (members == null)
            members = new LinkedList<Member>();

        if (name.equals(""))
            return false;

        for (Member item : members) {
            if (item.name.equals(name))
                return false;
        }
        Member m = new Member();
        m.name = name;
        m.value = 0;
        m.isWinner = false;
        members.add(m);
        return true;
    }

    public void deleteMember(List<Member> members, String selected) {
        for (Member item : members) {
            if (item.name.equals(selected)) {
                members.remove(item);
                return;
            }
        }
    }

    public void printMembersStat() {
        for (Member item : Storage.Members) {
            System.out.println(item.name + " " + item.value + " " + item.isWinner);
        }
    }
}
